openmdao/drivers/differential_evolution_driver.py

```diff
@@ -0,0 +1,483 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import copy
+import numpy as np
+import openmdao
+
+from openmdao.core.analysis_error import AnalysisError
+from openmdao.core.driver import Driver, RecordingDebugging
+from openmdao.utils.mpi import MPI
+from six import iteritems, itervalues, next
+
+try:
+    from tqdm import tqdm
+except ModuleNotFoundError:
+
+    def tqdm(i, _):
+        return i
+
+
+from differential_evolution import DifferentialEvolution, EvolutionStrategy
```

Comment:
I wanted to have differential_evolution be its own Python module so it can also be used outside of OpenMDAO as a pure optimizer. That is why I chose to import it as a module rather than include the code in the OpenMDAO repo.

Reply:
@Kenneth-T-Moore can you agree to this?

Reply:
I think this is fine, since it is important for someone to be able to import the differential optimizer for standalone usage. It probably even lends weight to the idea of keeping this as an external "plugin" in its entirety, but we will have to see where discussions take us at the OpenMDAO workshop.

Reply:
Alright, great! I will unfortunately not be able to make it to the workshop because of obligations at the office. But please let me know what comes up! :)